File: lindera-python/examples/tokenize.py
from lindera.dictionary import load_dictionary
from lindera.tokenizer import Tokenizer


def main():
    # load the dictionary
    dictionary = load_dictionary("embedded://ipadic")

    # create a tokenizer
    tokenizer = Tokenizer(dictionary, mode="normal")

    text = "関西国際空港限定トートバッグを東京スカイツリーの最寄り駅であるとうきょうスカイツリー駅で買う"
    print(f"text: {text}\n")

    # tokenize the text
    tokens = tokenizer.tokenize(text)

    for token in tokens:
        print(token.surface)


if __name__ == "__main__":
    main()

File: lindera-nodejs/examples/tokenize.js
const { loadDictionary, Tokenizer } = require("lindera-nodejs");

function main() {
  // load the dictionary
  const dictionary = loadDictionary("embedded://ipadic");

  // create a tokenizer
  const tokenizer = new Tokenizer(dictionary, "normal");

  const text =
    "関西国際空港限定トートバッグを東京スカイツリーの最寄り駅であるとうきょうスカイツリー駅で買う";
  console.log(`text: ${text}\n`);

  // tokenize the text
  const tokens = tokenizer.tokenize(text);

  for (const token of tokens) {
    console.log(token.surface);
  }
}

main();
